File: src/main/java/com/github/shatteredsuite/scrolls/data/scroll/UnboundBindingData.java
package com.github.shatteredsuite.scrolls.data.scroll;

import com.github.shatteredsuite.core.include.nbt.NBTCompound;
import com.github.shatteredsuite.scrolls.items.ScrollInstance;
import java.util.LinkedHashMap;
import java.util.Map;
import org.bukkit.entity.Player;

public class UnboundBindingData extends BindingData {

    public UnboundBindingData() {
        super("location");
    }

    @Override
    public ScrollInstance onInteract(ScrollInstance instance, Player player) {
        return new ScrollInstance(instance.scrollType, instance.charges, instance.isInfinite, new LocationBindingData(player.getLocation()));
    }

    @Override
    protected void applyBindingNBT(NBTCompound compound) {
        // Intentionally blank; no data to be added.
    }

    @Override
    public Map<String, Object> serialize() {
        return new LinkedHashMap<>();
    }
}

File: src/main/kotlin/com/github/shatteredsuite/scrolls/data/scroll/binding/UnboundBindingData.kt
package com.github.shatteredsuite.scrolls.data.scroll.binding

import com.github.shatteredsuite.core.include.nbt.NBTCompound
import com.github.shatteredsuite.scrolls.items.ScrollInstance
import org.bukkit.entity.Player
import java.util.*

class UnboundBindingData : BindingData("unbound", BindingDisplay("Unbound Scroll", false, LinkedList(), false, 0)) {
    override fun onInteract(instance: ScrollInstance, player: Player): ScrollInstance {
        return ScrollInstance(instance.scrollType, instance.charges, instance.isInfinite, LocationBindingData(player.location))
    }

    override fun applyBindingNBT(compound: NBTCompound) {
        // Intentionally blank; no data to be added.
    }

    override fun serialize(): Map<String?, Any?> {
        return LinkedHashMap()
    }
}